src/components/pages/teacher-info-page/TeacherInformPage.jsx

```diff
@@ -0,0 +1,197 @@
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchTeacherAndClasses = async () => {
+      try {
+        const teacherData = await getUserData(teacherId, userData.token);
+        setTeacherInfo(teacherData.data);
+
+        const response = await getClassesData();
+        const allClasses = response.classes;
+
+        const myClassIds = teacherData.data.myClasses.map((id) => id);
+        const filteredClasses = allClasses.filter((classItem) =>
+          myClassIds.includes(classItem._id)
+        );
+
+        if (filteredClasses.length === 0) {
+          setClassesError({
+            noClassesError: `This teacher hasn't added any classes yet.`,
+          });
+        } else {
+          setClasses(filteredClasses);
+        }
+
+        setIsLoading(false);
+      } catch (error) {
+        error('Error fetching data:', error);
+        setClassesError({
+          fetchError: 'Failed to fetch data. Please try again later.',
+        });
+        setIsLoading(false);
+      }
+    };
+
+    if (userData) {
+      fetchTeacherAndClasses();
+    } else {
+      error('No teacher data found');
+    }
+  }, [teacherId, userData]);
+
+  if (isLoading) return <Loader />;
+  if (classesError.fetchError) return <p>{classesError.fetchError}</p>;
+
+  return (
+    <div className="">
+      <div className="flex flex-col md:flex-row justify-center md:justify-around md:bg-lightGreen">
+        <div className="flex flex-col bg-pureWhite p-4 lg:p-10 ml-4 md:ml-12 md:my-12 rounded md:w-1/2">
+          <div className="flex flex-row gap-4">
+            <div className="flex m-2">
+              {teacherInfo?.profileImageUrl && (
+                <img
+                  src={teacherInfo.profileImageUrl}
+                  alt={`${teacherInfo.firstName} ${teacherInfo.lastName}`}
+                  className="w-20 h-20 rounded-full"
+                />
+              )}
+            </div>
+            <div className="flex justify-center items-center">
+              <h1 className="text-center font-roboto font-medium">
+                {teacherInfo?.firstName} {teacherInfo?.lastName}
+              </h1>
+            </div>
+          </div>
+
+          <div className="flex mt-4 gap-4">
+            <button className="font-spartan bg-red hover:bg-pureWhite hover:text-red px-4 py-1 border-2 border-transparent hover:border-red text-white font-spartan font-semibold text-sm sm:text-lg rounded-md transition duration-300 ease-in">
+              Book lesson
+            </button>
+            <button className="font-spartan bg-pureWhite px-4 py-1 hover:bg-red hover:text-pureWhite hover-border-red text-red font-spartan font-semibold text-lg rounded-md border-2 border-red">
+              Send Message
+            </button>
+          </div>
+        </div>
+        <div className="flex flex-col justify-center mx-auto p-4 m-4 md:m-12 rounded w-10/12 md:w-1/2 bg-lightGreen">
```

Comment:
We need to show here Profile Video
      
      
        <div className="flex flex-col justify-center mx-auto p-4 m-4 md:m-12 rounded w-10/12 md:w-1/2 bg-lightGreen">
          {teacherInfo?.profileVideoUrl && (
              <div className="bg-lightGreen rounded ">
                <video
                  className=" w-full rounded "
                  controls
                  src={teacherInfo.profileVideoUrl}
                  type="video/mp4"
                >
                  Your browser does not support the video format.
                </video>
              </div>
            )}
        </div>

Reply:
done